src/fsharp/vs/IncrementalBuild.fs

```diff
@@ -1901,5 +1901,5 @@ type IncrementalBuilder(ctokCtor: CompilationThreadToken, frameworkTcImportsCach
         | Some builder -> builder.IncrementUsageCount() 
```

Comment:
What happens if it is decremented to 0 between getting it, and incrementing the count?
Then it may have been disposed already, but this would set the counter to 1 again.

I think you would need a central routine, which increments the counter first, checks if it was > 0 before, then returns Some (or None if it was at zero).